### PR TITLE
CircleCI simplecov report merger for parallel specs

### PR DESCRIPTION
**Pull request by LindseySaari**
## Description of change
When running parallel specs in CircleCI, we get partial simplecov reports in each container, which causes the Simplecov minimum coverage to fail. This PR fetches all coverage jsons generated by simplecov across the various parallel containers of the Circle build, and merges them. The coverage files are stored as artifacts in the spec job for each node and the workspace is then passed to the subsequent coverage job. The coverage job runs after the spec job and runs a rake task `bundle exec rake simplecov:report_coverage` to merge the results.

## Original issue(s)
department-of-veterans-affairs/va.gov-team#13108

## Testing
- [x] The entire build passes in CircleCI
- [x] A simplecov report is generated per node and merged into one

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -32,18 +32,26 @@
...
 
+  defaults: &defaults
+    working_directory: ~/vets-api
+    executor: vets_api_docker_executor
+
   jobs:
     verify_pacts:
-      working_directory: ~/vets-api
+      <<: *defaults
       executor: vets_api_docker_executor
```
Comment (erluti) [inline .circleci/config.yml:42]: the defaults would also be pulling in the `executor` too, right?

Comment (LindseySaari) [reply .circleci/config.yml:42]: Yes, good catch! Forgot to remove

File: .circleci/config.yml [hunk 2]
```diff
@@ -32,54 +32,88 @@
...
+      <<: *defaults
       steps:
         - checkout_and_install
         - setup_test_db
         - verify_stable_pacts
         - store_test_results:
             path: ./rspec
+        - store_artifacts:
+            path: ./ci
+        - store_artifacts:
```
Comment (erluti) [inline .circleci/config.yml:50]: can this key be duplicated

Comment (LindseySaari) [reply .circleci/config.yml]: Yes, there's no limit on the number of store_artifacts steps a job can run. Actually in this case, I just cleaned up a couple of the jobs that don't store any artifacts at all. Thanks for pointing this out!

File: lib/tasks/simplecov_parallel.rake [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+require 'common/exceptions/base_error'
```
Comment (erluti) [inline lib/tasks/simplecov_parallel.rake:3]: is this using this?

**Review by erluti — APPROVED** [review]
LGTM

File: lib/tasks/simplecov_parallel.rake (hunk 3 above)
Comment (LindseySaari) [reply lib/tasks/simplecov_parallel.rake:3]: Removing it 👍

**Pull request merged**